crates/stages/src/stages/headers.rs

```diff
@@ -125,14 +201,8 @@ where
     ) -> Poll<Result<(), StageError>> {
         let current_checkpoint = input.checkpoint();
 
-        // Return if buffer already has some items.
-        if self.buffer.is_some() {
-            // TODO: review
-            trace!(
-                target: "sync::stages::headers",
-                checkpoint = %current_checkpoint.block_number,
-                "Buffer is not empty"
-            );
+        // Return if stage has already completed the gap
+        if self.is_etl_ready {
```

Comment:
isn't this check the same as `if gap.is_closed()`?

Reply:
added additional context  on the doc but no:
* `gap` checks the DB to see if the gap is closed
* `is_etl_ready` is true only when the ETL has all headers that will be inserted onto the DB